### ISSUE TITLE
Macro arguments in the middle of expressions.

### ISSUE THREAD

**Issue by yenatch** [gbdev/rgbds#63]
``` asm
A__ = -1

test_1: macro
    db \1_\1_\1
endm

test_2: macro
    db \1_\2_\3
endm

section "main", rom0

    test_1 A       ; -1
    test_2 A, B, C ; -1
```

Numbers are weirder:

``` asm
test_3: macro
    db \11
endm

test_4: macro
    db -\1
endm

test_5: macro
    db 1\12
endm

test_6: macro
    db 1\1
endm

section "main", rom0

    test_3 1       ; 11
    test_4 1       ; -1
    test_5 -       ; -1
    test_5 1       ; syntax error
    test_6 1       ; syntax error
```

**Comment by AntonioND** [comment]
I'm taking a look at this, but I'll post my findings so far:

For numbers, the problem is that the lexer, as soon as it finds a number, it looks for the first non-valid character for a number, sends it to the parser and continues. That way, when you do `1\1` the parser sees `1 \1`. However, when it's done the other way around, the lexer finds the first numeric value when the argument is replaced by the value of the argument. Only after finishing the argument it finds the other number, so it considers it a continuation.

For labels, it works just fine, at least with the current version.

Also, it seems that EQUSes can't be used in the list of macro arguments, the lexer used for macro arguments is a lot simpler than the one used for any other part of the code and interprets them as simple strings of characters that are copied when used from inside the macro. Macro arguments only allow a few escape characters (including macro arguments) as seen in `yylex_MACROARGS`. This has to be documented.

**Comment by ISSOtm** [comment]
State of this right now: the first block complains that `A_A_A` and `A_B_C` are not defined, respectively, so that's good.

The second block returns 11 (expected), -1 (expected), -1 (expected), a syntax error (oops), and another syntax error (oops)

**Comment by ISSOtm** [comment]
Fixed by #557.
